Document STATE_PATH and persistent state for Elastic Agent on Docker (#6504)
elastic/docs-content: reference/fleet/agent-environment-variables.md

| $$$env=common-kibana-username$$$<br>`KIBANA_USERNAME`<br> | (string) The basic authentication username used to connect to {{kib}} to retrieve a `service_token`.<br><br>**Default:** `elastic`<br> |
| $$$env=common-kibana-password$$$<br>`KIBANA_PASSWORD`<br> | (string) The basic authentication password used to connect to {{kib}} to retrieve a `service_token`.<br><br>**Default:** `changeme`<br> |
| $$$env-common-kibana-ca$$$<br>`KIBANA_CA`<br> | (string) The path to a certificate authority.<br><br>By default, {{agent}} uses the list of trusted certificate authorities (CA) from the operating system where it is running. If the certificate authority that signed your node certificates is not in the host system’s trusted certificate authorities list, use this config to add the path to the `.pem` file that contains your CA’s certificate.<br><br>**Default:** `""`<br> |

> | $$$env-common-elastic-netinfo$$$<br>`ELASTIC_NETINFO`<br> | (bool) When `false`, disables `netinfo.enabled` parameter of `add_host_metadata` processor. We recommend setting this to `false` for large-scale setups where the host.ip and host.mac fields index size increases.<br><br>By default, {{agent}} initializes the `add_host_metadata` processor. The `netinfo.enabled` parameter defines ingestion of IP addresses and MAC addresses as fields `host.ip` and `host.mac`. For more information see [add_host_metadata](/reference/fleet/add_host_metadata-processor.md)<br><br>**Default:** `"false"`<br> |

Reviewer: ```suggestion
| $$$env-common-elastic-netinfo$$$<br>`ELASTIC_NETINFO`<br> | (bool) When `false`, turns off the `netinfo.enabled` parameter of the `add_host_metadata` processor. Set this to `false` for large-scale setups where the `host.ip` and `host.mac` fields index size increases.<br><br>By default, {{agent}} initializes the `add_host_metadata` processor. The `netinfo.enabled` parameter defines ingestion of IP addresses and MAC addresses as fields `host.ip` and `host.mac`. For more information see [add_host_metadata](/reference/fleet/add_host_metadata-processor.md)<br><br>**Default:** `"false"`<br> |
```

| $$$env-bootstrap-fleet-fleet-server-elasticsearch-ca$$$<br>`FLEET_SERVER_ELASTICSEARCH_CA`<br> | (string) The path to a certificate authority file used to communicate with {{es}}. Overrides `ELASTICSEARCH_CA` when set. This is an alternative to using `FLEET_SERVER_ELASTICSEARCH_CA_TRUSTED_FINGERPRINT`. <br><br>By default, {{agent}} uses the list of trusted certificate authorities (CA) from the operating system where it is running. If the certificate authority that signed your node certificates is not in the host system’s trusted certificate authorities list, use this config to add the path to the `.pem` file that contains your CA's certificate.<br><br>**Default:** `""`<br> |
| $$$env-bootstrap-fleet-fleet-server-es-cert$$$<br>`FLEET_SERVER_ES_CERT`<br> | (string) The path to the mutual TLS client certificate that {{fleet-server}} will use to connect to {{es}}.<br><br>**Default:** `""`<br> |
| $$$env-bootstrap-fleet-fleet-server-es-cert-key$$$<br>`FLEET_SERVER_ES_CERT_KEY`<br> | (string) The path to the mutual TLS private key that {{fleet-server}} will use to connect to {{es}}.<br><br>**Default:** `""`<br> |

> | $$$env-bootstrap-fleet-fleet-server-insecure-http$$$<br>`FLEET_SERVER_INSECURE_HTTP`<br> | (bool) When `true`, {{fleet-server}} is exposed over insecure or unverified HTTP. We don't recommend setting this to `true`.<br><br>**Default:** `false`<br> |

Reviewer: ```suggestion
| $$$env-bootstrap-fleet-fleet-server-insecure-http$$$<br>`FLEET_SERVER_INSECURE_HTTP`<br> | (bool) When `true`, {{fleet-server}} is exposed over insecure or unverified HTTP. Don't set this to `true` unless required.<br><br>**Default:** `false`<br> |
```

I wonder if this should be documented at all.

Author: It's not like you can't see the setting though. Better to have it documented with a strong discouragement than leave no comment I guess.

| $$$env-enroll-fleet-token-name$$$<br>`FLEET_TOKEN_NAME`<br> | (string) The token name to use to fetch the token from {{kib}}.<br><br>**Default:** `""`<br> |
| $$$env-enroll-fleet-token-policy-name$$$<br>`FLEET_TOKEN_POLICY_NAME`<br> | (string) The token policy name to use to fetch the token from {{kib}}.<br><br>**Default:** `false`<br> |
| $$$env-enroll-fleet-ca$$$<br>`FLEET_CA`<br> | (string) The path to a certificate authority. Overrides `ELASTICSEARCH_CA` when set.<br><br>By default, {{agent}} uses the list of trusted certificate authorities (CA) from the operating system where it is running. If the certificate authority that signed your node certificates is not in the host system’s trusted certificate authorities list, use this config to add the path to the `.pem` file that contains your CA’s certificate.<br><br>**Default:** `false`<br> |

> | $$$env-enroll-fleet-insecure$$$<br>`FLEET_INSECURE`<br> | (bool) When `true`, {{agent}} communicates with {{fleet-server}} over insecure or unverified HTTP. We don't recommend setting this to `true`.<br><br>**Default:** `false`<br> |

Reviewer: ```suggestion
| $$$env-enroll-fleet-insecure$$$<br>`FLEET_INSECURE`<br> | (bool) When `true`, {{agent}} communicates with {{fleet-server}} over insecure or unverified HTTP. Don't set this to `true` unless required.<br><br>**Default:** `false`<br> |
```
